swap_meet/vendor.py

```diff
@@ -1,2 +1,69 @@
+
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except Exception:
+            return False
+
+    def get_by_category(self, category):
+        items_by_category = []
+
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+        
+        return items_by_category
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        self.inventory.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        other_vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        return True
+
+    def swap_first_item(self, other_vendor):
+        if self.inventory == [] or other_vendor.inventory == []:
+            return False
+
+
+        else:
+            self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+            return True
```

Comment:
Since the initial check above returns when the needed conditions (non-empty inventories) aren't met, we can treat it as a guard clause, removing the explicit else, and unindenting the following "main" logic.

```python
        self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
        return True
```